### ISSUE TITLE
Resources without models?

### ISSUE THREAD

**Issue by rlmartin** [mbryzek/apidoc#440]
I've been looking at a problem with the Swagger import, but I don't think it's unique to this. The Resource model requires a Model, presumably returned by a http 200 response on a GET (in most cases). Do you think it is a reasonable use case to not require the Model if all the Resource operations return http 204 (unit)? I know you hate booleans, but imagine an endpoint that toggles a value. The operations might be something like this:

GET: 204 (true), 404 (false) - gets the boolean value
PUT: 204 (success), 400 (error) - sets the boolean to true
DELETE: 204 (success), 400 (error) - sets the boolean to false

All actions and responses on this endpoint can be encapsulated using HTTP verbs and response codes - no Model required. What do you think of this? I can take a stab at sprinkling Options in the relevant places, but don't have a good sense of what might break down internally through the entire project structure.

**Comment by mbryzek** [comment]
Yeah - I've run into the same a few times. The core difference is the mindset shift between exposing resources as part of rest vs. allowing arbitrary API.

I don't think it would be too hard to model some kind of magic "there is no resource" - and can see this as useful esp when coming from a swagger definition

A current workaround
[hack.txt](https://github.com/mbryzek/apidoc/files/268919/hack.txt)

 is to create an 'empty' model and map that as the resource - e.g. just create a model that you don't use.

**Comment by rlmartin** [comment]
I'm going to put this aside for now. This would imply that Resource.type and Resource.plural both become Options - which leaves no required fields in the Resource model. There are a bunch of places where a unique reference to the Resource is required, and with no guarantee that any property would be set, it's not obvious how to identify a "name" for the Resource. Also, the UI would become polluted (something like a mix of Model names and URL paths in the sidebar). So without a clear vision of how to deal with this in the UI, it seems best to not mess with the existing format.

**Comment by mbryzek** [comment]
Example document demonstrating the use case attached
[test.txt](https://github.com/mbryzek/apidoc/files/764706/test.txt)

**Comment by mbryzek** [comment]
General concern that it feels weird to have a model w/ no fields... plan to discuss w/ ryan on next cell
